### PR TITLE
Remove C++ SSL Certificate API

### PR DESCRIPTION
**Pull request by pepone**
This PR removes the C++ Certificate API, `IceSSL::ConnectionInfo` provides the native peer certificate, (not a certificate chain).

### CODE REVIEW

File: cpp/include/Ice/SSL/Certificate.h
Comment (pepone) [inline cpp/include/Ice/SSL/Certificate.h:1]: I need to rename this header. 

- DistinguishedName can go into an internal header
- Exception would remain public.

File: cpp/include/Ice/SSL/ConnectionInfo.h [hunk 1]
```diff
@@ -19,43 +19,127 @@
...
-        ConnectionInfo& operator=(const ConnectionInfo&) = delete;
+        SchannelConnectionInfo(const SchannelConnectionInfo&) = delete;
+        SchannelConnectionInfo& operator=(const SchannelConnectionInfo&) = delete;
 
         /**
-         * The certificate chain.
+         * The peer certificate.
          */
-        std::vector<CertificatePtr> certs;
+        PCCERT_CONTEXT peerCertificate = nullptr;
```
Comment (bernardnormier) [inline cpp/include/Ice/SSL/ConnectionInfo.h:58]: Public fields is not a common / recommended style in C++.

It's ok for plain struct but not for such a class. It should be instead:
```cpp
public:
    PCCERT_CONTEXT peerCertificate() const noexcept { return _peerCertificate; }

private:
    const PCCERT_CONTEXT _peerCertificate = nullptr;
```

File: cpp/include/Ice/SSL/ConnectionInfo.h [hunk 2]
```diff
@@ -19,43 +19,127 @@
 
 namespace Ice::SSL
 {
+#if defined(ICE_USE_SCHANNEL)
     /**
      * Provides access to the connection details of an SSL connection.
      */
-    class ICE_API ConnectionInfo : public Ice::ConnectionInfo
+    class ICE_API SchannelConnectionInfo : public Ice::ConnectionInfo
```
Comment (bernardnormier) [inline cpp/include/Ice/SSL/ConnectionInfo.h:26]: + final?

File: cpp/include/Ice/SSL/ConnectionInfo.h [hunk 3]
```diff
@@ -19,43 +19,127 @@
...
+
+        ~SecureTransportConnectionInfo() override;
+
+        SecureTransportConnectionInfo(const SecureTransportConnectionInfo&) = delete;
+        SecureTransportConnectionInfo& operator=(const SecureTransportConnectionInfo&) = delete;
+
+        /**
+         * The peer certificate.
+         */
+        SecCertificateRef peerCertificate = nullptr;
```
Comment (bernardnormier) [inline cpp/include/Ice/SSL/ConnectionInfo.h:98]: Same here.

File: cpp/include/Ice/SSL/ConnectionInfoF.h [hunk 4]
```diff
@@ -5,12 +5,22 @@
...
+#include "Config.h"
+
 #include <memory>
 
 namespace Ice::SSL
 {
-    class ConnectionInfo;
-    using ConnectionInfoPtr = std::shared_ptr<ConnectionInfo>;
+#if defined(ICE_USE_SCHANNEL)
+    class SchannelConnectionInfo;
```
Comment (bernardnormier) [inline cpp/include/Ice/SSL/ConnectionInfoF.h:15]: I would include the aliasing here (using ConnectionInfo = SchannelConnectionInfo) and remove it from ConnectionInfo.h

**Review by bernardnormier — APPROVED** [review]
Lools good.

File: cpp/include/Ice/SSL/ConnectionInfo.h (hunk 1 above)
Comment (pepone) [reply cpp/include/Ice/SSL/ConnectionInfo.h:58]: The base class has public fields for other properties, should we fix them in a follow-up PR?

**Pull request merged**